cfme/infrastructure/host.py

```diff
@@ -92,15 +92,29 @@ def __init__(self, **kwargs):
             super(Host.Credential, self).__init__(**kwargs)
             self.ipmi = kwargs.get('ipmi')
 
-    def update(self, updates, validate_credentials=False):
+    def update(self, updates, validate_credentials=False, action='edit_from_details'):
         """Updates a host in the UI. Better to use utils.update.update context manager than call
         this directly.
 
         Args:
            updates (dict): fields that are changing.
+           action (str): denotes additional functionality. Expecting edit_from_details,
+           edit_from_hosts, delete, cancel, or nav_away.
```

Comment:
For `cancel`, we already have consistently used a specific boolean kwarg for this behavior within CRUD methods.

https://github.com/ManageIQ/integration_tests/search?q=%22cancel%3DFalse%22&unscoped_q=%22cancel%3DFalse%22

I don't think this update method should wrap that into the action - as its mutually exclusive with the navigation path used to get to update.

Reply:
We do have a cancel kwarg implemented for delete. I use this in the test. I don't see the same functionality for edit/update. Maybe I should implement it for update.